### DISCUSSION TITLE
best approach to put a serial number per each product

### DISCUSSION THREAD

**Discussion by developez** [vendurehq/vendure#1085] [Q&A]
What can be the best approach to put a serial number to product, is it feasible to uso one productvariant per "physic unit" of a product? (of course, each product has got only 1 of stock).

**Comment by kevmtt** [comment]
You can add a custom field to the ProductVariant to add a serial number.
https://www.vendure.io/docs/developer-guide/customizing-models/

**Comment by developez** [comment]
Yes, but a productvariant could has many items, but the serial number is unique per item.

**Comment by kevmtt** [comment]
Isn't a serial number something like a SKU? Just a identifier for a individual product?

When I understand you correctly you would have a product: `product 1`.
This `product 1` has many different variants based on the option group `physic unit`.
Then you create a product variant for each  `physic unit` value with a unique SKU (your serial number)
For each product variant you set the stock to be `1`.
